snuba/clickhouse/formatter/query.py

```diff
@@ -167,3 +168,36 @@ def format_simple_query(query: Query) -> FormattedQuery:
             if value
         ]
     )
+
+
+class JoinFormatter(JoinVisitor[FormattedNode, Table]):
+    """
+    Formats a Join tree.
+    """
+
+    def visit_individual_node(self, node: IndividualNode[Table]) -> FormattedNode:
+        """
+        An individual node is formatted as a table name with a table
+        alias, thus the padding.
+        """
+        return PaddingNode(None, format_data_source(node.data_source), node.alias)
+
+    def visit_join_clause(self, node: JoinClause[Table]) -> FormattedNode:
+        join_type = f"{node.join_type.value} " if node.join_type else ""
+        modifier = f"{node.join_modifier.value} " if node.join_modifier else ""
+        return SequenceNode(
+            [
+                node.left_node.accept(self),
+                StringNode(f"{join_type}{modifier}JOIN"),
+                node.right_node.accept(self),
+                StringNode("ON"),
+                SequenceNode(
```

Comment:
I'm not sure this will work, SequenceNode joins on a space, but according to the clickhouse docs the ON statements should be joined with commas.

Reply:
I think you are right. I did not test multiple conditions. Will verify

Reply:
Fixed. Actually the join conditions is a clickhouse conditions. Well, with a lot of limitations, in practice only AND. So these keys should be joined with an `AND`.

I think later on, we may consider extending the JoinCondition node not to be a simple mapping but actually an AST expression. That will be fairly easy to do.